tools/scripts/create-json-for-manifest.ps1

```diff
@@ -43,43 +43,81 @@ function CreatePaddedVersion([string]$version) {
     return $paddedVersion
 }
 
-function GetHighestBuiltVersion([string]$msiBasePath) {
-    Write-Verbose "  Entering GetHighestBuiltVersion"
+function Get-HighestBuiltVersion([string]$msiBasePath) {
+    Write-Verbose "  Entering Get-HighestBuiltVersion"
     Write-Verbose "    msiBasePath = $msiBasePath"
 
     $folders = Get-ChildItem -Path $msiBasePath -Directory | select Name
     $highestBuiltVersion = ($folders | Sort-Object -Property Name -Descending)[0].Name
 
-    Write-Verbose "  Exiting GetHighestBuiltVersion with $highestBuiltVersion"
+    Write-Verbose "  Exiting Get-HighestBuiltVersion with $highestBuiltVersion"
     return $highestBuiltVersion
 }
 
-function GetMsiInfo([string]$msiFilePath) {
-    Write-Verbose "  Entering GetMsiInfo"
+function Get-MsiInfo([string]$msiFilePath) {
+    Write-Verbose "  Entering Get-MsiInfo"
     Write-Verbose "    msiFilePath = $msiFilePath"
 
     $info = [ordered]@{}
     $info.msi_size_in_bytes = (Get-Item $msiFilePath).length
     $info.msi_sha_512 = (Get-FileHash -Path $msiFilePath -Algorithm "SHA512").Hash
 
-    Write-Verbose "  Exiting GetMsiInfo"
+    Write-Verbose "  Exiting Get-MsiInfo"
     return $info
 }
 
-function CreateJsonForManifest([string]$msiBasePath, [string]$minimumProductionVersion) {
+# Initialize the client
+function Get-Client()
+{
+    Write-Verbose "    Entering Get-Client"
+    # Load the octokit dll
+    $octoKitPath = "$($env:UserProfile)\.nuget\packages\Octokit\5.0.0\lib\netstandard2.0\Octokit.dll"
+    Write-Verbose "      Path to OctoKit.dll = $($octoKitPath)"
+    Add-Type -Path $($octoKitPath)
+
+    # Get a new client
+    $productHeader = [Octokit.ProductHeaderValue]::new("CreateJsonForManifest")
+    $client = [Octokit.GitHubClient]::new($productHeader)
+
+    Write-Verbose "    Exiting Get-Client"
+    return $client
+}
+
+function Get-MinimumProductionVersion([string]$currentVersion, $isMandatoryProdUpdate) {
+    Write-Verbose "  Entering Get-MinimumProductionVersion"
+    Write-Verbose "    currentVersion = $currentVersion"
+    Write-Verbose "    isMandatoryProdUpdate = $isMandatoryProdUpdate"
+
+    if ($isMandatoryProdUpdate -eq "true") {
+        Write-Verbose "    Using currentVersion for minimumProductionVersion"
+        $minimumProductionVersion = $currentVersion
+    } else {
+        Write-Verbose "    Fetching version from 'https://github.com/microsoft/accessibility-insights-windows/releases/tag/latest'"
+
+        $client = Get-Client
+        $release = $client.Repository.Release.GetLatest("Microsoft", "accessibility-insights-windows").Result
```

Comment:
Assuming the release object also contains error state and such? What happens if you attempt to access the `.result` member if an error occurs/the release can't be fetched?

Reply:
If an error occurs (can't reach the endpoint or somehow there's no "latest" build, then an Exception will get thrown and the script will fail